### PR TITLE
Erg #60 search bar list

### PR DESCRIPTION
**Pull request by derpdead**

### CODE REVIEW

File: components/List/ListSearchHeader.vue [hunk 1]
```diff
@@ -58,29 +61,35 @@ export default {
...
-            this.$emit('searchResult', this.searchResult);
-        },
+    beforeDestroy() {
+        delete this.debouncedSearch;
     },
     computed: {
+        searchBtnIconComponent() {
+            return this.isSearchBtnClicked
+                ? () => import('~/components/Icon/Window/IconClose')
+                : IconSearch;
```
Comment (bleto) [inline components/List/ListSearchHeader.vue]: Why you use synchronous and asynchronous import on this property? use asynchronous.

File: model/mappers/gridDataMapper.js [hunk 2]
```diff
@@ -214,3 +214,31 @@ export function getMappedFilter(filter) {
 
     return mappedFilter;
 }
+
+export function getMappedElementsToGroups(elements) {
+    const { length } = elements;
+    const groupedElements = {};
```
Comment (bleto) [inline model/mappers/gridDataMapper.js]: it should be array

Comment (derpdead) [reply model/mappers/gridDataMapper.js]: its map with key values as arrays

File: components/List/ListSearchHeader.vue (hunk 1 above)
Comment (derpdead) [reply components/List/ListSearchHeader.vue]: It's no matter if it is sync nor async component in computed

**Pull request merged**